redash/models/parameterized_query.py

```diff
@@ -88,6 +89,16 @@ def _is_number(string):
         return True
 
 
+def _is_regex_pattern(value, regex):
+    try:
+        if re.compile(regex).match(value):
```

Comment:
Let's use [fullmatch()](https://docs.python.org/3/library/re.html#re.fullmatch) instead so that the pattern has to cover the entire string.